Sources/ComposableArchitecture/Reducer.swift

```diff
@@ -360,44 +483,59 @@ public struct Reducer<State, Action, Environment> {
   ///     `GlobalState`.
   ///   - toLocalAction: A case path that can extract/embed `(Int, Action)` from `GlobalAction`.
   ///   - toLocalEnvironment: A function that transforms `GlobalEnvironment` into `Environment`.
+  ///   - breakpointOnNil: Raises `SIGTRAP` signal when an action is sent to the reducer but the
+  ///     index is out of bounds. This is generally considered a logic error, as a child reducer
+  ///     cannot process a child action for unavailable child state.
```

Comment:
I dunno if `breakpointOnNil` is the right name for the `forEach` reducers...maybe we can come up with something better.